homeassistant/components/smhi/config_flow.py

```diff
@@ -38,7 +38,7 @@ async def async_check_location(
 class SmhiFlowHandler(ConfigFlow, domain=DOMAIN):
     """Config flow for SMHI component."""
 
-    VERSION = 2
+    VERSION = 3
```

Comment:
This prevents downgrade. Is it necessary?
I think it's easier to simply stop adding the CONF_NAME, and never read it, without actually removing it from existing entries.

Reply:
Agree, that would be better 👍

Reply:
In my opinion, we need to update the version and prevent downgrades as the change is backward incompatible.
Only users who have set up the integration before this PR is merged can downgrade.
Imagine a user sets the integration up after this PR is merged and, therefore, has no `NAME` in the config entry. Then he downgrades, and we will see some KeyErrors in his log instead that the integration is not loaded.

Reply:
I chatted with @joostlek about it, and for him, it's fine to keep the name and not update the version as the no-used data will not harm.
So, I would say the integration owner decides whether to clean up the unused values or keep them.
As we don't support downgrade in general, we don't need to make sure that downgrade is supported

Reply:
I gave it some thought from your previous comment and I will change this back again to bump version for two reasons.
- It's more clean and better experience for the user with migration error rather than a key error
- Remove not needed data ( sure, it's nothing in the whole context, but every byte counts 🤣 )